lockable/resource_filter.py

```diff
@@ -0,0 +1,43 @@
+""" Resource filter module """
+
+import re
+
+from pydash import filter_
+
+
+def get_matching_resources(resource_list: [{}], requirements: {}) -> [{}]:
+    """ Get matching resources from resource list """
+    def in_filter(resource: dict):
+        """ Check if resource matches requirements """
+        for key, value in requirements.items():
+            if isinstance(value, dict):
+                # check that dictionary contains only accepted keys
+                accepted_keys = ['$exists', '$in', '$nin', '$regex']
+                for k in value.keys():
+                    assert k in accepted_keys, f'Unsupported key: {k} in {value}'
```

Comment:
this would block nested filtering usage now . need some more work to support also nested search..